### PR TITLE
[new release] ppx_distr_guards (0.3)

### PR DESCRIPTION
**Pull request by vogler**
Extension to distribute guards over or-patterns

- Project page: <a href="https://github.com/vogler/ppx_distr_guards">https://github.com/vogler/ppx_distr_guards</a>
- Documentation: <a href="https://vogler.github.io/ppx_distr_guards">https://vogler.github.io/ppx_distr_guards</a>

##### CHANGES:

- Migrate from ocaml-migrate-parsetree to ppxlib.

### CODE REVIEW

File: packages/ppx_distr_guards/ppx_distr_guards.0.3/opam [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+description:
+  "`function%distr A x, _ | _, A x when p x -> e` will result in `function A x, _ when p x -> e | _, A x when p x -> e`"
+maintainer: ["Ralf Vogler <ralf.vogler@gmail.com>"]
+authors: ["Ralf Vogler <ralf.vogler@gmail.com>"]
+license: "MIT"
+homepage: "https://github.com/vogler/ppx_distr_guards"
+doc: "https://vogler.github.io/ppx_distr_guards"
+bug-reports: "https://github.com/vogler/ppx_distr_guards/issues"
+depends: [
+  "dune" {>= "2.7" & >= "2.3.0"}
```
Comment (mseri) [inline packages/ppx_distr_guards/ppx_distr_guards.0.3/opam:12]:
```suggestion
  "dune" {>= "2.7"}
```

**Comment by kit-ty-kate** [comment]
Thanks! Could you return the fix in the opam file upstream so they can be taken into account for future releases?

File: packages/ppx_distr_guards/ppx_distr_guards.0.3/opam (hunk 1 above)
Comment (vogler) [reply packages/ppx_distr_guards/ppx_distr_guards.0.3/opam]: This got generated from the [dune-project](https://github.com/vogler/ppx_distr_guards/blob/0ec3a096baa6f147c8981eb1f2d11fb72f865f2a/dune-project#L17) file.
If I adjust the constraint there to `(dune (>= 2.7))` the generated opam constraint is `"dune" {>= "2.7" & >= "2.7"}`. I assume it adds `2.7` from `(lang dune 2.7)`.
Should this be reported somewhere or do you just manually adjust or ignore it (or leave out the dune constraint)?

**Comment by vogler** [comment]
> Thanks! Could you return the fix in the opam file upstream so they can be taken into account for future releases?

See https://github.com/ocaml/opam-repository/pull/18981#discussion_r661861502.
Not sure how this should be handled. It will generate the opam file like that every build.

**Comment by kit-ty-kate** [comment]
You just need to remove the `(dune (>= 2.3.0))` line from your dune-project file and call `dune build` to regenerate the opam file.

File: packages/ppx_distr_guards/ppx_distr_guards.0.3/opam (hunk 1 above)
Comment (kit-ty-kate) [reply packages/ppx_distr_guards/ppx_distr_guards.0.3/opam]: This is a dune issue. See https://github.com/ocaml/dune/pull/3705

**Comment by vogler** [comment]
Ok, works. Thanks!

**Pull request merged**